runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java

```diff
@@ -127,5 +169,135 @@ private void setDodgyCharges(int dodgyCharges)
 	{
 		this.dodgyCharges = dodgyCharges;
 		config.dodgyNecklace(dodgyCharges);
+		if (currentInfoBoxIDs.get(EquipmentInventorySlot.AMULET) == ItemID.DODGY_NECKLACE)
+		{
+			if (dodgyCharges > 0)
+			{
+				createJewelleryInfobox(ItemID.DODGY_NECKLACE, dodgyCharges, EquipmentInventorySlot.AMULET);
+			}
+			else
+			{
+				removeJewelleryInfobox(EquipmentInventorySlot.AMULET);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(ItemContainerChanged event)
+	{
+		if (event.getItemContainer().getItems().length > 14)
+		{
+			return;
+		}
+
+		Item[] items = event.getItemContainer().getItems();
+
+		int amulet = -2;
+		int bracelet = -2;
+		int ring = -2;
+		if (items.length >= 3)
+		{
+			if (config.showAmuletInfobox())
+			{
+				amulet = items[EquipmentInventorySlot.AMULET.getSlotIdx()].getId();
+			}
+			if (items.length >= 10)
+			{
+				if (config.showBraceletInfobox())
```

Comment:
You can make these checks more linear by inlining the item.length checks

Reply:
With the current flow, I can't, as it will stop rings from being shown if the user does not want to see bracelets

Reply:
`if (config.showBraceletInfobox() && items.length >= 10)`

and

`if (config.showRingInfobox() && items.length >= 13)